src/Microsoft.Extensions.Logging/LoggerRuleSelector.cs

```diff
@@ -47,9 +49,22 @@ private static bool IsBetter(LoggerFilterRule rule, LoggerFilterRule current, st
                 return false;
             }
 
-            if (rule.CategoryName != null && !category.StartsWith(rule.CategoryName, StringComparison.OrdinalIgnoreCase))
+            if (rule.CategoryName != null)
             {
-                return false;
+                var categoryParts = rule.CategoryName.Split(WildcardChar);
+                if (categoryParts.Length > 2)
+                {
+                    throw new InvalidOperationException("Only one wildcard character is allowed in category name.");
+                }
+
+                var prefix = categoryParts[0];
+                var suffix = categoryParts.Length > 1 ? categoryParts[1] : string.Empty;
```

Comment:
Should we enforce the invariant that a wildcard needs to replace whole segments?

Right now this supports things like `Micro*` or `*soft` which just doesn't make sense.